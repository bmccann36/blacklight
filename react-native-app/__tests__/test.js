import 'react-native';
import React from 'react';
import { connect } from 'react-redux';
// import AR from '../components/AR';
import Login from '../components/Login';
// import Main from '../components/Main';
// import Map from '../components/Map';
import SingleMemoryView from '../components/SingleMemoryView';
// import RouterComponent from '../Router';
// let store;


import { createStore } from 'redux';
import { Button } from 'react-native-elements'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<SingleMemoryView />).toJSON();
  expect(tree).toMatchSnapshot();
});


/* Check Button */
describe('Login component', () => {
  it('renders without crashing', () => {
    const renderedButton = renderer.create(<Button onPress={() => {}} />).toJSON();
    expect(renderedButton).toMatchSnapshot();
  })
})


// test('renders correctly', () => {
//   const tree = renderer.create(<RouterComponent />).toJSON();
//   expect(tree).toMatchSnapshot();
// });



// test('renders correctly', () => {
//   const tree = renderer.create(<Map />).toJSON();
//   expect(tree).toMatchSnapshot();
// });

// test('renders correctly', () => {
//   const tree = renderer.create(<AR />).toJSON();
//   expect(tree).toMatchSnapshot();
// });

// test('renders correctly', () => {
//   const tree = renderer.create(<Login /> store={store}).toJSON();
//   expect(tree).toMatchSnapshot();
// });

// test('renders correctly', () => {
//   const tree = renderer.create(<Main />).toJSON();
//   expect(tree).toMatchSnapshot();
// });

// jest.mock('../components/Main', () => {
//   const mockComponent = require('react-native/jest/Main');
//   return mockComponent('../components/Main');
// });

it('works', () => {
  expect(1).toBe(1);
});



//auth_test.js

// import 'react-native';
// import React from 'react';
// import {auth, INITIAL_STATE} from '../store/auth';
// // import {Login, onPasswordChange } from '../components/Login';
// import {user, initialState, createUserSuccess} from '../store/user';

// expect(auth(INITIAL_STATE, onPasswordChange(text))).toEqual({
//   email: '', password: text, user: {}
// })

// expect(user(initialState, createUserSuccess(user))).toEqual({

// })
