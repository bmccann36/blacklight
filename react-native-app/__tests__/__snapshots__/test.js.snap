// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 15,
        "marginRight": 15,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      pointerEvents="box-only"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9E9E9E",
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 19,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "padding": 12,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontSize": 17.5,
              },
              undefined,
              undefined,
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`Login component renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 15,
        "marginRight": 15,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      pointerEvents="box-only"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9E9E9E",
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 19,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "padding": 12,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontSize": 17.5,
              },
              undefined,
              undefined,
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "flex": 1,
      "flexDirection": "row",
      "flexWrap": "wrap",
      "height": undefined,
      "justifyContent": "center",
      "padding": 20,
      "width": undefined,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "height": undefined,
        "justifyContent": "center",
        "padding": 20,
        "width": undefined,
      }
    }
  >
    <RCTScrollView>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "#b20000",
              "fontSize": 22,
              "fontWeight": "bold",
              "marginTop": 30,
              "padding": 10,
              "textShadowColor": "#000000",
              "textShadowOffset": Object {
                "height": 5,
                "width": 5,
              },
              "textShadowRadius": 8,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "#FFFFFF",
              "fontSize": 20,
              "marginTop": 30,
              "padding": 10,
              "textShadowColor": "#000000",
              "textShadowOffset": Object {
                "height": 5,
                "width": 5,
              },
              "textShadowRadius": 8,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
</View>
`;
